### PR TITLE
[ADD] merge_models method

### PR DESCRIPTION
**Pull request by MiquelRForgeFlow**
Based on https://github.com/OCA/OpenUpgrade/pull/3173.

### CODE REVIEW

File: openupgradelib/openupgrade.py [hunk 1]
```diff
@@ -917,6 +918,104 @@ def rename_models(cr, model_spec):
     # TODO: signal where the model occurs in references to ir_model
 
 
+def merge_models(cr, old_model, new_model, ref_field):
+    """
+    Update model references for models that have merged to an existing model.
+    :param old_model: old model
+    :param new_model: destination model
+    :param ref_field: name of the field in new model that references old model.
```
Comment (StefanRijnhart) [inline openupgradelib/openupgrade.py:926]: I would prefer if this mentioned that it refers to the `id` in the old model's table (if that is indeed the case).

File: openupgradelib/openupgrade.py [hunk 2]
```diff
@@ -917,6 +918,104 @@ def rename_models(cr, model_spec):
...
+            ('mail_message_subtype', 'res_model', '', ''),
+            ('mail_activity', 'res_model', 'res_id', 'res_model_id'),
+            ('mail_activity_type', '', '', 'res_model_id'),
+            ('mail_template', 'model', '', 'model_id'),
+            ('mail_alias', '', '', 'alias_model_id'),
+            ('mail_alias', '', 'alias_parent_thread_id',
+             'alias_parent_model_id'),
+            # mail_followers: special case handled below
+        ]
+    for rename in renames:
```
Comment (StefanRijnhart) [inline openupgradelib/openupgrade.py:951]: I think the code would be more clear if you replaced `rename` with a tuple, perhaps something like
```
for (table, model_name_column, res_id_column, model_id_column) in renames:
```

File: openupgradelib/openupgrade.py [hunk 3]
```diff
@@ -917,6 +918,104 @@ def rename_models(cr, model_spec):
...
+            UPDATE {table} t
+            SET """ + query_1a + query_3 + query_2a + """
+            WHERE """ + query_1b + query_2b
+        logged_query(cr, sql.SQL(query).format(
+            model_table=sql.Identifier(table),
+            table=sql.Identifier(rename[0]),
+            field1=sql.Identifier(rename[1]),
+            field2=sql.Identifier(rename[2]),
+            field3=sql.Identifier(rename[3]),
+            field=sql.Identifier(ref_field)
```
Comment (StefanRijnhart) [inline openupgradelib/openupgrade.py:990]: Here too, the query would be more clear to me I think if the `field1`, `field2`(etc) keys were renamed to something like `model_name_column` etc. What do you think?

File: openupgradelib/openupgrade.py (hunk 2 above)
Comment (MiquelRForgeFlow) [reply openupgradelib/openupgrade.py:951]: Ok, I like your suggestion. I will proceed with it.

**Comment by MiquelRForgeFlow** [comment]
@StefanRijnhart done

**Review by leemannd — APPROVED** [review]

**Review by StephaneMangin — APPROVED** [review]

**Review by sebalix — APPROVED** [review]

**Pull request merged**